internal/provider/provider.go

```diff
@@ -60,7 +68,11 @@ func configure(version string, p *schema.Provider) func(context.Context, *schema
 
 		var diags diag.Diagnostics
 
-		controllerAddress := d.Get("controller").(string)
```

Comment:
If the type of `controller_addresses` was `schema.TypeList`, we should be able to obtain the addresses without parsing like this.